### ISSUE TITLE
Custom input component or element

### ISSUE THREAD

**Issue by gpoole** [Hacker0x01/react-datepicker#529]
What do you think about supporting customising the actual `<input />` by allowing the user to pass in their own? Something like:

```jsx
<DatePicker input={InputComponent} />
<DatePicker input={<InputComponent someCustomProp />} />
```

I think it's fairly common to have a base `Input` component which provides extra classes, styles or functionality, so it's useful to be able to use that here too. Obviously whatever's passed in would have to support the props already being passed to the regular `input`.

**Comment by maxkrieger** [comment]
I was very surprised to see this not already available considering most people use this for the picker itself. It's pretty compromising for accessibility when you're stuck with a dummy textbox and can't use an `<input type="date">` or just a more custom solution/button.

**Comment by gpoole** [comment]
Fixed in #551!

**Comment by bORm** [comment]
<CustomInputComponent onClick={ this.props.onClick } .... 
add toggle Visibility calendar
